Signal/src/util/ThreadUtil.m

```diff
@@ -361,7 +369,15 @@ + (ThreadDynamicInteractions *)ensureDynamicInteractionsForThread:(TSThread *)th
             shouldHaveBlockOffer = NO;
         }
 
+        if (![OWSProfileManager.sharedManager hasLocalProfile] ||
+            [OWSProfileManager.sharedManager isThreadInProfileWhitelist:thread]) {
+            // Don't show offer if thread is local user hasn't configured their profile.
+            // Don't show offer if thread is already in profile whitelist.
+            shouldHaveAddToProfileWhitelistOffer = NO;
+        }
```

Comment:
I think we talked about periodically re-showing this nag (e.g. once a month).  I haven't done that yet.